index.js

```diff
@@ -978,7 +982,8 @@ class Wire extends Duplex {
     }
     // now this._buffer is an array containing a single Buffer
     if (this._cryptoSyncPattern) {
-      const index = this._buffer[0].indexOf(this._cryptoSyncPattern)
+      // convert to Buffer for finding arrays of bytes with indexOf
+      const index = Buffer.from(this._buffer[0]).indexOf(this._cryptoSyncPattern)
```

Comment:
dont use buffer, use Uint8Array

Reply:
Here I used Buffer because I wanted the whole cryptoSync array to match, not just the first byte. TypedArray.indexOf cannot find the index of a match for an entire array. Buffer can do that natively. AFAIK if we wanted to use Uint8Array to find an entire array, we would need to write a new indexOfMulti function ([example](https://github.com/aleixrodriala/wa-tunnel/blob/main/src/utils/string-utils.js#L13)), but I think it would be better to export that from the uint8-util library rather than put it in here.

Reply:
then do that, but if you use Buffer this PR will never be merged